app/src/main/java/com/github/ybecker/epforuml/SavedQuestionsFragment.kt

```diff
@@ -13,11 +20,59 @@ import android.view.ViewGroup
  * create an instance of this fragment.
  */
 class SavedQuestionsFragment : Fragment() {
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var user : Model.User
+
+    private lateinit var cache : ArrayList<Model.Question>
+    private lateinit var newIntentDetails : Intent
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
+        cache = this.requireArguments().getParcelableArrayList("savedQuestions")!!
+
         // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_saved_questions, container, false)
     }
+
+    @SuppressLint("SetTextI18n")
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // initialize recycler with saved questions
+        user = DatabaseManager.user ?: Model.User()
+        val userId = user.userId
+
+        if (userId.isNotEmpty() && cache.isNotEmpty()) {
+            val layoutManager = LinearLayoutManager(context)
+            recyclerView = view.findViewById(R.id.recycler_saved_questions)
+            recyclerView.layoutManager = layoutManager
+
+            // display saved questions
+            val adapter = ForumAdapter(cache)
+            recyclerView.adapter = adapter
+
+            // move to QuestionDetails when clicking on specific question
+            adapter.onItemClick = {q ->
+                newIntentDetails = Intent(context?.applicationContext, QuestionDetailsActivity::class.java)
```

Comment:
By going in question details you use the AnswerAdapter, which uses the database itself, so it will not work offline. That's why you should implement a new Question adapter that uses the cache